plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/client/OpenSearchClient.java

```diff
@@ -382,33 +386,35 @@ public List<String> getIndexes()
     {
         return doRequest("/_cat/indices?h=index,docs.count,docs.deleted&format=json&s=index:asc", body -> {
             try {
-                ImmutableList.Builder<String> result = ImmutableList.builder();
                 JsonNode root = OBJECT_MAPPER.readTree(body);
+                List<String> indices = new ArrayList<>();
+
                 for (int i = 0; i < root.size(); i++) {
-                    String index = root.get(i).get("index").asText();
-                    // make sure the index has mappings we can use to derive the schema
-                    int docsCount = root.get(i).get("docs.count").asInt();
-                    int deletedDocsCount = root.get(i).get("docs.deleted").asInt();
-                    if (docsCount == 0 && deletedDocsCount == 0) {
-                        try {
-                            // without documents, the index won't have any dynamic mappings, but maybe there are some explicit ones
-                            if (getIndexMetadata(index).schema().fields().isEmpty()) {
-                                continue;
-                            }
+                    indices.add(root.get(i).get("index").asText());
+                }
+
+                return indices.parallelStream().map(index -> {
```

Comment:
We usually don't use `parallelStream` in this project. I would recommend focusing on the permission issue in this PR.